### PR TITLE
Combine sign up, sign in and sign out feature tests

### PR DESCRIPTION
**Pull request by tijmenb**
### Context

In https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/246 we introduced a new style for feature specs. Not all existing specs were updated in that PR. 

### Changes proposed in this pull request

This rewrites the acceptance tests into a single flow that walks through the flow of a user:

- signing up
- signing out
- signing in again
- signing in via the sign up form

I didn't want to add the token-checking code in the integration test, so I've combined these with the existing token specs in a single request spec.

### Guidance to review

- Does the big spec make sense?
- Are all relevant scenarios covered?

### Link to Trello card

N/A

### CODE REVIEW

File: spec/system/candidate_interface/candidate_account_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.feature 'Candidate account' do
+  scenario 'Candidate signs up, out, and in again' do
+    given_i_am_a_candidate_without_an_account
+
+    when_i_visit_the_signup_page
+    and_i_submit_my_email_address
+    then_i_receive_an_email_with_a_signin_link
```
Comment (RobertM905) [inline spec/system/candidate_interface/candidate_account_spec.rb]: Should this not be a sign up link?
Are sign up and sign in email content going to be the same?

File: spec/requests/candidate_interface/authentication_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+RSpec.describe 'Authentication for candidates', type: :request do
+  it 'redirects the user if the token does not exist' do
+    get candidate_interface_application_form_url(token: '123')
+
+    # TODO: add a better check once we have implemented error messages
+    expect(response).to have_http_status(302)
+  end
+
+  it 'redirects the user if the token is missing' do
```
Comment (alexdesi) [inline spec/requests/candidate_interface/authentication_spec.rb]: '... token is missing'  and 'token does not exist' (from the previous test) sounds very similar, you may want to make it clearer.
I think you want to say:
- the token param is missing
- maybe the second means that token is invalid (hence does not exist)

File: spec/system/candidate_interface/candidate_account_spec.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    when_i_click_the_sign_out_button
+
+    when_i_visit_the_signup_page
+    and_i_submit_my_email_address
+    when_i_click_on_the_link_in_my_email
+    then_i_am_signed_in
+
+    when_i_signed_in_more_than_a_week_ago
+    then_i_should_be_signed_out
+  end
```
Comment (alexdesi) [inline spec/system/candidate_interface/candidate_account_spec.rb]: this test is massive 🐘 :)  
smol(ler) would be nice(r) 🐭

File: spec/system/candidate_interface/candidate_account_spec.rb (hunk 1 above)
Comment (tijmenb) [reply spec/system/candidate_interface/candidate_account_spec.rb]: Updated!

**Pull request merged**